torchgeo/datasets/seasonet.py

```diff
@@ -0,0 +1,456 @@
+    ]
+    image_size = (120, 120)
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        seasons: Collection[str] = all_seasons,
+        bands: Iterable[str] = all_bands,
+        grids: Iterable[int] = [1, 2],
+        concat_seasons: int = 1,
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new SeasoNet dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train", "val" or "test"
+            seasons: list of seasons to load
+            bands: list of bands to load
+            grids: which of the overlapping grids to load
+            concat_seasons: number of seasonal images to return per sample.
+                if 1, each seasonal image is returned as its own sample,
+                otherwise seasonal images are randomly picked from the seasons
+                specified in `seasons` and returned as stacked tensors
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+        """
+        assert split in self.splits
+        for season in seasons:
+            assert season in self.all_seasons
+        for band in bands:
+            assert band in self.all_bands
+        for grid in grids:
+            assert grid in [1, 2]
+        assert concat_seasons in range(1, len(seasons) + 1)
+
+        self.root = root
+        self.bands = bands
+        self.concat_seasons = concat_seasons
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+        self.cmap = ListedColormap(np.array(self.colormap) / 255)
+
+        self._verify()
+
+        self.channels = 0
+        for b in bands:
+            self.channels += self.band_nums[b]
+
+        csv = pd.read_csv(os.path.join(self.root, "meta.csv"), index_col="Index")
+
+        if split is not None:
+            split_csv = pd.read_csv(
+                os.path.join(self.root, f"splits/{split}.csv"), header=None
+            )[0]
+            csv = csv.iloc[split_csv]
+
+        csv = csv[csv["Grid"].isin(grids)]
+        csv = csv[csv["Season"].isin(seasons)]
+        csv["Path"] = csv["Path"].apply(
+            lambda p: [os.path.join(self.root, p, os.path.basename(p))]
+        )
+
+        if self.concat_seasons > 1:
+            self.files = csv.groupby(["Latitude", "Longitude"])
+            self.files = self.files["Path"].agg(sum)
+            self.files = self.files[
+                self.files.apply(lambda d: len(d) >= self.concat_seasons)
+            ]
+        else:
+            self.files = csv["Path"]
```

Comment:
Might be useful to add comments to `__init__`, at first glance I have no idea what all of this pandas logic is doing.